Latest nvidia driver doesn't compile

**Describe the bug**
With the latest kernel and the latest non-beta nvidia driver, the drivers don't compile:
```
error: builder for '/nix/store/wv00lqc6ynx5bhhv31vaixaz867a9vfb-nvidia-x11-460.73.01-5.13.drv' failed with exit code 2;
       last 10 log lines:
       > make[3]: *** Waiting for unfinished jobs....
       > /build/kernel/nvidia-drm/nvidia-drm-modeset.c: In function '__will_generate_flip_event':
       > /build/kernel/nvidia-drm/nvidia-drm-modeset.c:96:23: warning: unused variable 'primary_plane' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
       >    96 |     struct drm_plane *primary_plane = crtc->primary;
       >       |                       ^~~~~~~~~~~~~
       > make[2]: *** [/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/source/Makefile:1847: /build/kernel] Error 2
       > make[2]: Leaving directory '/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/build'
       > make[1]: *** [Makefile:215: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/source'
       > make: *** [Makefile:80: modules] Error 2
       For full logs, run 'nix log /nix/store/wv00lqc6ynx5bhhv31vaixaz867a9vfb-nvidia-x11-460.73.01-5.13.drv'.
error: 1 dependencies of derivation '/nix/store/gld36fh5sc69s3lw8rnpm701pbsxx67g-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5qnd3i9k96nkxljjs6hpiq100k9ss1m6-linux-5.13-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/11y6fqa7ygg9bbr9mf2rq61ciq6dyzk1-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l29n869335p9xfyvcaqbjgqwc8pkcwl5-nixos-system-dracula-21.11.20210703.09c38c2.drv' failed to build
error: getting status of '/tmp/nixos-rebuild.v0uAIA/result': No such file or directory
```

**To Reproduce**
Steps to reproduce the behavior:
1. use unstable
2. use the latest linux kernel
3. add nvidia as a drivers

**Expected behavior**
It should compile.


**Notify maintainers**
Latest person who touched it: @ronthecookie 

**Metadata**
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.12, NixOS, 21.11.20210703.09c38c2 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `no` # also happens with it on, it has to be off for other reasons
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `""`
 - channels(nix): `""`
 - nixpkgs: `/nix/store/zy56rp0mfp7lx2xqcfzqcf9z82vph9pd-wsfwxm10shvi4yr0ll8p4v1dp7l7wqfl-source`
```

--- a/pkgs/os-specific/linux/nvidia-x11/default.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/default.nix
@@ -19,10 +19,10 @@ rec {
   # Policy: use the highest stable version as the default (on our master).
   stable = if stdenv.hostPlatform.system == "x86_64-linux"
     then generic {
-      version = "460.73.01";
-      sha256_64bit = "120ymf59l6nipczszf82lrm2p4ihhqyv2pfwwfg9wy96vqcckc8i";
-      settingsSha256 = "08jh7g34p9yxv5fh1cw0r2pjx65ryiv3w2lk1qg0gxn2r7xypkx0";
-      persistencedSha256 = "040gx4wqp3hxcfb4aba4sl7b01ixr5slhzw0xldwcqlmhpwqphi5";
+      version = "470.57.02";
+      sha256_64bit = "sha256-VdeuEEgn+qeel1Mh/itg+d1C+/9lZCBTRDwOVv20xH0=";
+      settingsSha256 = "sha256-DJg5QbyuKJmPpLQVYgTLvucI1e9YgQOO16690VXIWvk=";
+      persistencedSha256 = "sha256-Cqv6oUFnsSi3S1sjplJKeq9bI2pqgBXPPb11HOJSlDo=";
     }
     else legacy_390;
 

